books/RayTracingTheRestOfYourLife.html

```diff
@@ -423,184 +571,504 @@
 </div>
 
 <div class='together'>
-This says _the probability that a random variable with our PDF is less than one is 25%_. This gives
-rise to a clever observation that underlies many methods to generate non-uniform random numbers. We
-want a function `f()` that when we call it as `f(random_double())` we get a return value with a PDF
-$\frac{x^2}{4}$. We don’t know what that is, but we do know that 25% of what it returns should be
-less than 1.0, and 75% should be above 1.0. If $f()$ is increasing, then we would expect $f(0.25) =
-1.0$. This can be generalized to figure out $f()$ for every possible input:
+If we evaluate the CDF, $P(r)$, at $r = 1.0$, we get:
 
-  $$ f(P(x)) = x $$
+    $$ P(1.0) = \frac{1}{4} $$
+
+This says _a random variable plucked from our PDF has a 25% change of being 1 or lower_. We want a
+function $f(d)$ that takes a uniform distribution between 0 and 1 (_i.e_ `f(random_double())`), and
+returns a random value according to a distribution that has the CDF $P(x) = \frac{x^2}{4}$. We don’t
+know yet know what the function $f(d)$ is analytically, but we do know that 25% of what it returns should
+be less than 1.0, and 75% should be above 1.0. Likewise, we know that 50% of what it returns should
+be less than $\sqrt{2}$, and 50% should be above $\sqrt{2}$. If $f(d)$ monotonically increases, then
+we would expect $f(0.25) = 1.0$ and $f(0.5) = \sqrt{2}$. This can be generalized to figure out
+$f(d)$ for every possible input:
+
+    $$ f(P(x)) = x $$
 </div>
 
+
 <div class='together'>
-That means $f$ just undoes whatever $P$ does. So,
+Let's take some more samples:
+
+    $$ P(0.0) = 0 $$
+    $$ P(0.5) = \frac{1}{16} $$
+    $$ P(1.0) = \frac{1}{4} $$
+    $$ P(1.5) = \frac{9}{16} $$
+    $$ P(2.0) = 1 $$
+
+so, the function $f()$ has values
+
+    $$ f(P(0.0)) = f(0) = 0 $$
+    $$ f(P(0.5)) = f(\frac{1}{16}) = 0.5 $$
+    $$ f(P(1.0)) = f(\frac{1}{4}) = 1.0 $$
+    $$ f(P(1.5)) = f(\frac{9}{16}) = 1.5 $$
+    $$ f(P(2.0)) = f(1) = 2.0 $$
+
+We could use these intermediate values and interpolate between them to approximate $f(d)$:
+
+    ![Figure [approx f]: Approximating the nonuniform f()](../images/fig-3.08-approx-f.jpg)
 
-  $$ f(x) = P^{-1}(x) $$
+If you can't solve for the PDF analytically, then you can't solve for the CDF analytically. After
+all, the CDF is just the integral of the PDF. However, you can still create a distribution that
+approximates the PDF. If you take a bunch of samples from the random function you want the PDF from,
+you can approximate the PDF by getting a histogram of the samples and then converting to a PDF.
+Alternatively, you can do as we did above and sort all of your samples.
 </div>
 
 <div class='together'>
+Looking closer at the equality:
+
+    $$ f(P(x)) = x $$
+
+That just means that $f()$ just undoes whatever $P()$ does. So,
+
+    $$ f(d) = P^{-1}(x) $$
+
 The -1 means “inverse function”. Ugly notation, but standard. For our purposes, if we have PDF $p()$
```

Comment:
I'd assume this is commonly known. If you still wish to keep it, perhaps make it a parenthetical comment.

Reply:
It was in the original text. I'll noodle of keeping/tossing